### PR TITLE
Ansible arg spec validation using AnsibleModule

### PR DESCRIPTION
**Pull request by cidrblock**
From an action
```
def _check_argspec(self):
    aav = AnsibleArgSpecValidator(
        data=self._task.args,
        schema=DOCUMENTATION,
        schema_format="doc",
        schema_conditionals={},
        name=self._task.action,
    )
    valid, errors = aav.validate()
    if not valid:
        raise AnsibleActionFail(errors)
```

### CODE REVIEW

File: plugins/module_utils/common/argspec_validate.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+import re
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.ansible.utils.plugins.module_utils.common.utils import (
+    dict_merge,
+)
+from ansible.module_utils.six import iteritems, string_types
+from ansible.module_utils._text import to_bytes
+
+try:
+    from packaging.specifiers import SpecifierSet
```
Comment (ganeshrn) [inline plugins/module_utils/common/argspec_validate.py:35]: Why is this import required?

Comment (cidrblock) [reply plugins/module_utils/common/argspec_validate.py]: It's not, left over.  I was originally using the ansible version to determine is the future ansible-base functionality should be used once it's out of AnsibleModule.  In the end I went with a try/except instead.  good catch

File: plugins/module_utils/common/argspec_validate.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+    try:
+        from yaml import CSafeLoader as SafeLoader
+    except ImportError:
+        from yaml import SafeLoader
+    HAS_YAML = True
+except ImportError:
+    HAS_YAML = False
+
+try:
```
Comment (ganeshrn) [inline plugins/module_utils/common/argspec_validate.py:46]: Can you please add the intention of having this import maybe as a `TODO:` section

File: plugins/module_utils/common/argspec_validate.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    def __init__(
+        self,
+        data,
+        schema,
+        schema_format,
+        schema_conditionals=None,
+        name=None,
+        other_args=None,
+    ):
+        self._errors = ""
```
Comment (ganeshrn) [inline plugins/module_utils/common/argspec_validate.py:141]: Can we document the argument list for sake of clarity?

**Review by ganeshrn — APPROVED** [review]
Looks good, apart from minor docs comment.

File: plugins/module_utils/common/argspec_validate.py (hunk 3 above)
Comment (cidrblock) [reply plugins/module_utils/common/argspec_validate.py:141]: good call, yes

File: plugins/module_utils/common/argspec_validate.py (hunk 2 above)
Comment (cidrblock) [reply plugins/module_utils/common/argspec_validate.py:46]: yep, good idea
